File: problems/042-first-duplicate-test/solutions/python/main.py
import sys


def input_tokens():
    for line in sys.stdin.buffer:
        yield from line.split()


tokens = input_tokens()
n = int(next(tokens))
first_index = {}
for index in range(1, n + 1):
    fingerprint = next(tokens)
    earliest = first_index.get(fingerprint)
    if earliest is not None:
        print(index, earliest)
        break
    first_index[fingerprint] = index
else:
    print("NONE")

File: problems/042-first-duplicate-test/solutions/rust/main.rs
use std::collections::HashMap;
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut tokens = input.split_whitespace();
    let n: usize = tokens.next().unwrap().parse().unwrap();
    let mut first_index: HashMap<&str, usize> = HashMap::with_capacity(n * 2);

    for index in 1..=n {
        let fingerprint = tokens.next().unwrap();
        if let Some(&earliest) = first_index.get(fingerprint) {
            println!("{} {}", index, earliest);
            return;
        }
        first_index.insert(fingerprint, index);
    }
    println!("NONE");
}
